internal/stage_a2.go

```diff
@@ -21,7 +21,14 @@ func testA2(stageHarness *test_case_harness.TestCaseHarness) error {
 
 	inputArgsAndCompletion := []inputArgsAndCompletion{
 		{Input: "ech", Completion: "echo", CompletionEndsWithNoSpace: false, Args: []string{"hello"}, Response: "hello"},
-		{Input: "exi", Completion: "exit", CompletionEndsWithNoSpace: false, Args: []string{"0"}, Response: ""},
+		{Input: "typ", Completion: "type", CompletionEndsWithNoSpace: false, Args: []string{"type"}, Response: "type is a shell builtin"},
```

Comment:
Looks like we're just substituting exit for type here. A user could very well complain that our tests are broken since we don't test completing exit? I think we should just test all the builtins from the base stages. 

Second point -- `type type` could be a bit confusing in logs, especially if a user has a broken autocompletion output that prints the command twice or so. I'd do `type echo` instead.